dataset/src/commonMain/kotlin/org/jetbrains/kotlinx/dl/dataset/ImagenetUtils.kt

```diff
@@ -0,0 +1,1040 @@
+package org.jetbrains.kotlinx.dl.dataset
+
+/**
+ * Type of Imagenet dataset labels.
+ * Some models are trained on the vanilla ImageNet dataset, which contains 1000 classes,
+ * but some models also output additional 'background' class.
+ */
+public enum class Imagenet {
+    V1k,
+    V1001;
+
+    public fun labels(zeroIndexed: Boolean = true) : Map<Int, String> {
+        return when (this) {
+            V1k -> if (!zeroIndexed) toOneIndexed(imagenetLabels) else imagenetLabels
+            V1001 -> if (!zeroIndexed) toOneIndexed(addBackgroundLabel(imagenetLabels)) else
+                addBackgroundLabel(imagenetLabels)
+        }
+    }
+
+    private fun toOneIndexed(labels: Map<Int, String>) : Map<Int, String> {
+        val zeroIndexedLabels = mutableMapOf<Int, String>()
+        labels.forEach { (key, value) ->
+            zeroIndexedLabels[key + 1] = value
+        }
+        return zeroIndexedLabels
+    }
+
+    private fun addBackgroundLabel(labels: Map<Int, String>) : Map<Int, String> {
+        val labelsWithBackground = mutableMapOf<Int, String>()
+        labelsWithBackground[0] = "background"
+        labels.forEach { (key, value) ->
+            labelsWithBackground[key + 1] = value
+        }
+        return labelsWithBackground
+    }
+}
+
+
+public val imagenetLabels: Map<Int, String> = mapOf(
```

Comment:
These are the same labels as loaded by `org.jetbrains.kotlinx.dl.api.inference.imagerecognition.ImageRecognitionModelKt.loadImageNetClassLabels`, right? Perhaps we should get rid of the function and use this enum everywhere?